plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -2542,26 +2544,33 @@ public void testAllAvailableTypes()
     @Test
     public void testLocalDynamicFilteringWithSelectiveBuildSizeJoin()
     {
-        long fullTableScan = (Long) computeActual("SELECT count(*) FROM lineitem").getOnlyValue();
+        // We need to prepare tables for this test. The test is required to use tables that are backed by at lest two files
+        Session session = Session.builder(getSession())
+                .setSystemProperty(TASK_WRITER_COUNT, "2")
```

Comment:
I don't think writer count >= 2 guarantees number of files >= 2.
It still may be possible that all data ends up on a single machine and single thread (just not very likely).

Explicit `partitioning`, or using small target-file-size can guarantee 2+ files.

Also, if the test relies on # of files, it should explicitly validate the state of the table, eg via `SELECT count(DISTINCT "$path")`.

cc @alexjo2144